### PR TITLE
Feat/budget planning

### PR DESCRIPTION
**Pull request by t-eliks**

### CODE REVIEW

File: DataAccess/Repository.cs [hunk 1]
```diff
@@ -7,6 +7,7 @@ public class Repository : DbContext
     {
         public Repository(DbContextOptions<Repository> options) : base(options)
         {
+            this.Database.EnsureCreated();
```
Comment (EdotJ) [inline DataAccess/Repository.cs:10]: I assume it's good to keep this here?

File: Web/Controllers/Category/CategoryController.cs [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+        }
+        
+        [HttpGet]
+        public IActionResult OpenCreationForm()
+        {
+            return View("CategoryForm", new CategoryViewModel());
+        }
+        
+        [HttpPost]
+        public IActionResult Create(CategoryViewModel viewModel)
```
Comment (EdotJ) [inline Web/Controllers/Category/CategoryController.cs:80]: According to MD, shouldn't this be Submit?

Comment (EdotJ) [reply Web/Controllers/Category/CategoryController.cs:80]: Same for Edit. Although that's a bs diagram, we should change that.

File: Web/Views/Budget/PlanPage.cshtml [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+                <select class="form-control"
+                        data-bind="options: categoryOptions, 
+                                                optionsValue: function(item) { return item.id }, 
+                                                optionsText: function(item) { return item.name; }, 
+                                                optionsCaption: 'Pasirinkite kategoriją',
+                                                value: categoryId">
+                </select>
+                <input data-bind="value: goal" class="form-control"/>
+                <div class="mb-2">
+                    <button class="btn btn-primary" data-bind="click: saveGoal">Išsaugoti</button>
```
Comment (EdotJ) [inline Web/Views/Budget/PlanPage.cshtml]: Any way to fix the bug where I can press save 10 times and it would add 10 new rows (when adding new row)

**Review by EdotJ — APPROVED** [review]
Looks good to my untrained C# eye, aside from the minor stuff

File: DataAccess/Repository.cs (hunk 1 above)
Comment (t-eliks) [reply DataAccess/Repository.cs:10]: As it says, it ensures that a database is created if one is not found at the connection string. Useful for quick database publishing.

File: Web/Views/Budget/PlanPage.cshtml (hunk 3 above)
Comment (t-eliks) [reply Web/Views/Budget/PlanPage.cshtml]: Fixed

**Pull request merged**